src/Adapter/Product/Image/Uploader/ProductImageUploader.php

```diff
@@ -169,19 +180,51 @@ private function deleteCachedImages(Image $image): void
         ];
 
         foreach ($cachedImages as $cachedImage) {
-            if (file_exists($cachedImage)) {
-                try {
-                    unlink($cachedImage);
-                } catch (ErrorException $e) {
-                    throw new CannotUnlinkImageException(
-                        sprintf(
-                            'Failed to remove cached image "%s"',
-                            $cachedImage
-                        ),
-                        0,
-                        $e
-                    );
-                }
+            if (!file_exists($cachedImage)) {
+                continue;
+            }
+
+            if (!@unlink($cachedImage)) {
+                throw new CannotUnlinkImageException(
+                    sprintf(
+                        'Failed to remove cached image "%s"',
+                        $cachedImage
+                    )
+                );
+            }
+        }
+    }
+
+    /**
+     * Note: we can't delete the whole folder here, or Image::delete will return an error
+     * because it expects the original image and the folder to be present in order to remove
+     * them correctly. So for now this service only handles removing generated image types.
+     * When Image ObjectModel is no longer used, it could also remove the remaining files.
+     *
+     * @param string $imagePath
+     * @param array $imageTypes
+     *
+     * @throws CannotUnlinkImageException
+     */
+    private function deleteGeneratedImages(string $imagePath, array $imageTypes): void
+    {
+        $fileExtension = pathinfo($imagePath, PATHINFO_EXTENSION);
+        $destinationExtension = '.jpg';
```

Comment:
I suppose the destinationExtension is the same when the images are generated. Can we have a constant which defines that extension so it will be easier if we want to change it in the future ?

Reply:
It should be dynamic actually, we kept it hardcoded here because it's a relicate from legacy We don't use const on purpose because it would imply we assume that it will always be jpg
It should be refacto in the future to handle other image types